### PR TITLE
GH-1995 push release branch and automate PR creation

### PR DESCRIPTION
**Pull request by jeenbroekstra**
- push release branch to avoid direct commits on master
- use hub command to create PR for release branch to master
- use hub command to create PR for updating develop branch

GitHub issue resolved: #1995

### CODE REVIEW

**Comment by jeenbroekstra** [comment]
I will merge this immediately so I can use it for the 3.1.2 release. Only way to really debug, unfortunately, is to actually use it for a release.

**Comment by hmottestad** [comment]
For testing i have previously forked the RDF4J repo.

File: release.sh [hunk 1]
```diff
@@ -153,7 +165,7 @@ git pull
 
 MVN_VERSION_DEVELOP=$(xmllint --xpath "//*[local-name()='project']/*[local-name()='version']/text()" pom.xml)
 
-git checkout master
+git checkout ${BRANCH}
```
Comment (hmottestad) [inline release.sh]: Isn’t this the branch we are on already?

File: release.sh [hunk 2]
```diff
@@ -13,6 +13,12 @@ increment_version() {
  echo "$v" | perl -pe s/$rgx.*$'/${1}'`printf %0${#val}s $(($val+1))`/
 }
 
+echo ""
+echo "The release script requires several external command line tools:"
+echo " - git"
+echo " - mvn"
+echo " - hub (https://hub.github.com/)"
```
Comment (hmottestad) [inline release.sh]: Can we use the official command line client instead? https://cli.github.com/

Comment (hmottestad) [reply release.sh]: Or is the benefit of using hub that we don’t have to log in again?

Comment (hmottestad) [reply release.sh]: Btw. We need to test that hub (or whatever we end up with) works as an early step in the script. So it doesn’t crash half way through.

Comment (jeenbroekstra) [reply release.sh]: I wasn't aware there was another client, I thought `hub` _is_ the official client. Happy to consider alternatives if they're better.

File: release.sh (hunk 1 above)
Comment (jeenbroekstra) [reply release.sh]: Yes, good point

File: release.sh (hunk 2 above)
Comment (jeenbroekstra) [reply release.sh]: > Or is the benefit of using hub that we don’t have to log in again?

No and actually that brings up a point: you need to set this tool up first before it works.

**Pull request merged**